sdk/storage_datalake/src/properties.rs

```diff
@@ -14,6 +14,12 @@ impl Default for Properties {
     }
 }
 
+impl From<BTreeMap<Cow<'static, str>, Cow<'static, str>>> for Properties {
+    fn from(value: BTreeMap<Cow<'static, str>, Cow<'static, str>>) -> Self {
+        Self(value)
+    }
+}
```

Comment:
Thought about providing a more flexible implementation, like

```rs
impl<T> From<T> for Properties
where
    T: Into<BTreeMap<Cow<'static, str>, Cow<'static, str>>>,
{
    fn from(value: T) -> Self {
        Self(value.into())
    }
}
```
however that conflicts with the `TryFrom<&Headers>` implementation, which needs to look into a specific header key. This seems to be related to some generic implementation in core - this was a bit surprising to me, since its different traits. but did not dig any deeper...